### PR TITLE
convert custom try macro to `?`

### LINKED ISSUE #60580: t! macro can be replaced by '?' operator
This file defines a `t!` macro, but it seems like we can remove that macro and just use the new `?` unary operator instead.

https://github.com/rust-lang/rust/blob/7ac02005f3d7ac4b2c32b82dde365107aa2aafdf/src/libstd/sys/unix/process/process_unix.rs#L170-L175

There might be other files with a `t!` macro defined.

### PR DESCRIPTION
**Pull request by hellow554**
resolves #60580

r? @frewsxcv

### CODE REVIEW

**Comment by hellow554** [comment]
Quick question: is the `AWS_ACCESS_KEY_ID` supposed to be visible? @Centril

File: src/libstd/sys/redox/process.rs [hunk 1]
```diff
@@ -218,7 +218,10 @@ impl Command {
         }
 
         match self.setup_io(default, true) {
-            Ok((_, theirs)) => unsafe { self.do_exec(theirs) },
+            Ok((_, theirs)) => unsafe {
+                let Err(e) = self.do_exec(theirs);
+                e
```
Comment (hellow554) [inline src/libstd/sys/redox/process.rs]: I first did `self.do_exec(theirs).err().unwrap()`. Is there a strong opinion about this? The neat one about this is static assertion

**Comment by mati865** [comment]
It might be not worth just now: https://github.com/rust-lang/rust/issues/37939

**Comment by hellow554** [comment]
@mati865 I'm sadly aware of that issue, but a quick test showed that since 1.32.0 they produce the same asm at least on godbolt

**Comment by Centril** [comment]
> Quick question: is the `AWS_ACCESS_KEY_ID` supposed to be visible? @Centril

@pietroalbini ^--

**Comment by pietroalbini** [comment]
> Quick question: is the AWS_ACCESS_KEY_ID supposed to be visible?

Yep, the access key id can be public. The actual sensitive part is `AWS_SECRET_ACCESS_KEY` which is not visible, and also not included in PR builds. We *could* make `AWS_ACCESS_KEY_ID` secret, but it would be annoying for us when we need to track down where a key is used.

**Comment by hellow554** [comment]
Oh boy.. what happened here? AFAIk I haven't closed this PR.

**Comment by hellow554** [comment]
[Concern by @mati865](https://github.com/rust-lang/rust/issues/60580#issuecomment-491244958)

> #37939 probably should be fixed first

**Comment by hellow554** [comment]
ping @frewsxcv

**Comment by hellow554** [comment]
@jonas-schievink can you reassign this PR please? @frewsxcv seems to be not available.

**Pull request merged**